src/System.Collections/src/System/Collections/Generic/Queue.cs

```diff
@@ -392,20 +388,48 @@ public bool MoveNext()
             {
                 if (_version != _q._version) throw new InvalidOperationException(SR.InvalidOperation_EnumFailedVersion);
 
-                if (_index == -2)
-                    return false;
-
-                _index++;
+                // Instead of return true/false, we assign the result to a variable and
+                // return it at the end. This helps decrease code size, as currently the
+                // jit cannot do this for us and generates code for 3 returns.
+                bool result = false;
```

Comment:
As previously commented, is this really saving enough / providing enough value to be worth churning the code, making it harder to read, etc.?  What measurement are you using?